src/WebJobs.Script.SiteExtension/New-PrivateSiteExtension.ps1

```diff
@@ -0,0 +1,130 @@
+<#
+    .SYNOPSIS
+    Produces a private site extension.
+
+    .DESCRIPTION
+    Takes in a published site extension and produces a private site extension.
+
+    .PARAMETER InputPath
+    The path of the published 'SiteExtension'. Leave null to scan for root in a child from here.
+
+    .PARAMETER OutputPath
+    The path to produce the private site extension to. When zipping, this is the name of the zip file. Leave null to compute this name.
```

Comment:
> Leave null to compute this name.

When I leave this empty and provide the `-Zip` switch, the script fails for me

```
> pwsh New-PrivateSiteExtension.ps1 -Zip
Processing -File 'New-PrivateSiteExtension.ps1' failed: Access to the path is denied. Specify a valid path for the -File parameter.
```

But providing the path worked for me i.e.

`pwsh New-PrivateSiteExtension.ps1 -OutputPath /Users/likasem/Downloads/drop_windows/privatesiteext -Zip`

Reply:
@liliankasem I think this is how you are invoking `pwsh` from your terminal. `New-PrivateSiteExtension.ps1 -Zip` works fine for my on windows.

Can you launch a pwsh terminal and invoke this directly?

Reply:
Ahh fun mac things; doing as you suggested works fine:

```
> pwsh
PowerShell 7.4.5
PS /Users/likasem/Downloads/drop_windows> ./New-PrivateSiteExtension.ps1 -Zip
Published private site extension to Functions.Private.4.1037.0-ci.25058.2.win-x86.zip
```

Reply:
My only other note on this script is that 9/10 times if you're getting a private site extension you would want to have it in a zip to deploy it. I think it makes more sense for -zip to be the default behavior here (nit, not blocking)

Reply:
Defaulted to zipping, replaced switch to `-NoZip`